x-pack/plugins/ml/public/application/components/data_grid/column_chart.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC } from 'react';
+
+import { BarSeries, Chart, Settings } from '@elastic/charts';
+import { EuiDataGridColumn } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import './column_chart.scss';
+
+import {
+  isNumericChartData,
+  isOrdinalChartData,
+  useColumnChart,
+  ChartData,
+} from './use_column_chart';
+
+const getLegendText = (chartData: ChartData, MAX_CHART_COLUMNS: number): string | JSX.Element => {
+  if (chartData.type === 'boolean') {
+    return (
+      <table className="mlDataGridChart__legendBoolean">
+        <tbody>
+          <tr>
+            <td>{chartData.data[0].key_as_string}</td>
+            <td>{chartData.data[1].key_as_string}</td>
+          </tr>
+        </tbody>
+      </table>
+    );
+  }
+
+  if (isOrdinalChartData(chartData) && chartData.cardinality === 1) {
+    return i18n.translate('xpack.ml.dataGridChart.singleCategoryLegend', {
+      defaultMessage: `{cardinality} category`,
+      values: { cardinality: chartData.cardinality },
+    });
+  }
+
+  if (isOrdinalChartData(chartData) && chartData.cardinality > MAX_CHART_COLUMNS) {
+    return i18n.translate('xpack.ml.dataGridChart.topCategoriesLegend', {
+      defaultMessage: `top {MAX_CHART_COLUMNS} of {cardinality} categories`,
+      values: { cardinality: chartData.cardinality, MAX_CHART_COLUMNS },
+    });
+  }
+
+  if (
+    isOrdinalChartData(chartData) &&
+    chartData.cardinality <= MAX_CHART_COLUMNS &&
+    chartData.cardinality > 1
+  ) {
+    return i18n.translate('xpack.ml.dataGridChart.categoriesLegend', {
+      defaultMessage: `{cardinality} categories`,
+      values: { cardinality: chartData.cardinality },
+    });
+  }
+
+  if (isNumericChartData(chartData)) {
+    return `${Math.round(chartData.stats[0] * 100) / 100} - ${
+      Math.round(chartData.stats[1] * 100) / 100
+    }`;
+  }
+
+  throw new Error('Invalid chart data.');
+};
+
+interface Props {
+  chartData: ChartData;
+  columnType: EuiDataGridColumn;
+}
+
+export const ColumnChart: FC<Props> = ({ chartData, columnType }) => {
+  const { coloredData, xScaleType, MAX_CHART_COLUMNS } = useColumnChart(chartData, columnType);
+
+  if (coloredData.length === 0) {
+    return columnType.id;
+  }
+
+  return (
+    <>
+      <div className="mlDataGridChart__histogram">
+        <Chart>
```

Comment:
The tooltip is not positioning correctly when the bar is on the left of the grid:

![image](https://user-images.githubusercontent.com/7405507/84652974-6d7f2300-af04-11ea-9c65-f65b5a2190cf.png)

Reply:
I asked the charts team about this and added a note to the meta issue https://github.com/elastic/kibana/issues/69235

Reply:
The tooltip positioning is now fixed by the latest EUI Charts library being available f9f1158e1f4dec2b125e936b61704a2dee122f78